File: java/src/com/jhobot/commands/fun/CatFact.java
package com.jhobot.commands.fun;

import com.jhobot.core.ChadBot;
import com.jhobot.handle.JSONHandler;
import com.jhobot.handle.MessageHandler;
import com.jhobot.handle.commands.*;
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageReceivedEvent;

import java.io.IOException;
import java.util.HashMap;
import java.util.List;

public class CatFact implements Command {
    @DefineCommand(category = Category.FUN)
    @Override
    public Runnable run(MessageReceivedEvent e, List<String> args) {
        return () -> {
            try{
                String fact = ChadBot.JSON_HANDLER.read("https://catfact.ninja/fact").getString("fact");
                new MessageHandler(e.getChannel()).send(fact, "Cat Fact");
            } catch (Exception ee)
            {
                ee.printStackTrace();
                new MessageHandler(e.getChannel()).sendError("There was an internal error.");
            }
        };
    }

    @Override
    public Runnable help(MessageReceivedEvent e, List<String> args) {
        HashMap<String, String> st = new HashMap<>();
        st.put("catfact", "Gives you a random catfact.");
        return HelpHandler.helpCommand(st, "Cat Fact", e);
    }
}

File: chad/src/main/java/dev/shog/chad/commands/fun/CatFact.kt
package dev.shog.chad.commands.`fun`

import dev.shog.chad.framework.handle.JsonHandler
import dev.shog.chad.framework.handle.MessageHandler
import dev.shog.chad.framework.obj.Command
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageEvent
import java.util.*

/**
 * Gets a random cat fact from an API
 *
 * @author sho
 */
class CatFact : Command.Class {
    override suspend fun run(e: MessageEvent, args: MutableList<String>) {
        JsonHandler.read("https://catfact.ninja/fact")!!
                .getString("fact")
                .also { MessageHandler(e.channel, e.author).credit("catfact.ninja").sendEmbed { withDesc(it) } }
    }

    override suspend fun help(e: MessageEvent) {
        val st = HashMap<String, String>()
        st["catfact"] = "Gives you a random catfact."
        Command.helpCommand(st, "Cat Fact", e)
    }
}
